syncer/dbconn/db.go

```diff
@@ -0,0 +1,234 @@
+package dbconn
+
+import (
+	"database/sql"
+	"strings"
+	"time"
+
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/tidb-tools/pkg/dbutil"
+	"go.uber.org/zap"
+
+	"github.com/pingcap/dm/dm/config"
+	"github.com/pingcap/dm/pkg/conn"
+	tcontext "github.com/pingcap/dm/pkg/context"
+	"github.com/pingcap/dm/pkg/log"
+	"github.com/pingcap/dm/pkg/retry"
+	"github.com/pingcap/dm/pkg/terror"
+	"github.com/pingcap/dm/pkg/utils"
+	"github.com/pingcap/dm/syncer/metrics"
+)
+
+var retryTimeout = 3 * time.Second
+
+// CreateBaseDB creates a db from config.
+func CreateBaseDB(dbCfg config.DBConfig) (*conn.BaseDB, error) {
+	db, err := conn.DefaultDBProvider.Apply(dbCfg)
+	if err != nil {
+		return nil, terror.WithScope(err, terror.ScopeDownstream)
+	}
+	return db, nil
+}
+
+// CloseBaseDB closes baseDB to release all connection generated by this baseDB and this baseDB.
+func CloseBaseDB(logCtx *tcontext.Context, baseDB *conn.BaseDB) {
+	if baseDB != nil {
+		err := baseDB.Close()
+		if err != nil {
+			logCtx.L().Error("fail to close baseDB", log.ShortError(err))
+		}
+	}
+}
+
+// DBConn represents a live DB connection
+// it's not thread-safe.
+type DBConn struct {
+	Cfg      *config.SubTaskConfig
+	BaseConn *conn.BaseConn
```

Comment:
Why do we need to export these fields?

Reply:
syncer need to access `BaseConn.DBConn` sometimes